docs/development/logging.md

```diff
@@ -0,0 +1,165 @@
+
+```go
+var logger logr.Logger
+logger.Info("scaling deployment", "deployment", client.ObjectKeyFromObject(deployment), "replicas", replicaCount)
+```
+
+## Log Configuration
+
+Gardener components can be configured to either log in `json` (default) or `text` format:
+`json` format is supposed to be used in production, while `text` format might be nicer for development.
+
+```text
+# json
+{"level":"info","ts":"2021-12-16T08:32:21.059+0100","msg":"hello botanist","garden":"eden"}
+
+# text
+2021-12-16T08:32:21.059+0100    INFO    hello botanist  {"garden": "eden"}
+```
+
+Components can be set to one of the following log levels (with increasing verbosity): `error`, `info` (default), `debug`.
+
+> ℹ️ Note: some Gardener components don't feature a configurable log level and format yet.
+> In this case they log at `info` in `json` format.
+> We might add configuration options via command line flags that can be used in all components in the future though.
+
+## Log Levels
+
+logr uses [V-levels](https://github.com/go-logr/logr#why-v-levels) (numbered log levels), higher V-level means higher verbosity.
+V-levels are relative (in contrast to `klog`'s absolute V-levels), i.e., `V(1)` creates a logger, that is one level more verbose than its parent loggger.
+
+In Gardener components, `V(0)` of the root logger maps to `info` level, `V(1)` means `debug` level.
+
+There is no `warning` level (see [Dave Cheney's post](https://dave.cheney.net/2015/11/05/lets-talk-about-logging)).
+If there is an error condition (e.g., unexpected error received from a called function), the error should either be handled or logged at `error` if it is neither handled nor returned.
+If you have an `error` value at hand that doesn't represent an actual error condition, but you still want to log it as an informational message, log it at `info` level with key `err`.
+
+We might consider to make use of a broader range of log levels in the future when introducing more logs and common command line flags for our components (comparable to `--v` of Kubernetes components).
+For now, we stick to the mentioned two log levels like controller-runtime: info (`V(0)`) and debug (`V(1)`).
+
+## Logging in Controllers
+
+### Named Loggers
+
+Controllers should use named loggers that include their name, e.g.:
+```go
+controllerLogger := rootLogger.WithName("controller").WithName("shoot")
+controllerLogger.Info("Deploying kube-apiserver")
+```
+results in
+```text
+2021-12-16T09:27:56.550+0100    INFO    controller.shoot    Deploying kube-apiserver
+```
+Logger names are hierarchical. You can make use of it, where controllers are composed of multiple "subcontrollers", e.g., `controller.shoot.hibernation` or `controller.shoot.maintenance`.
+
+Using the global logger `logf.Log` directly is discouraged and should be rather exceptional, because it makes correlating logs with code harder.
```

Comment:
```suggestion
Using the global logger `logf.Log` directly is discouraged and should be rather exceptional because it makes correlating logs with code harder.
```